packages/react-devtools-shared/src/backend/agent.js

```diff
@@ -754,6 +754,9 @@ export default class Agent extends EventEmitter<{
   shutdown: () => void = () => {
     // Clean up the overlay if visible, and associated events.
     this.emit('shutdown');
+
+    this._bridge.removeAllListeners();
```

Comment:
Is this safe? What happens if other objects are listening to the `shutdown` event too?

Reply:
It is safe, because `Agent` is actually notified by `Bridge` here:
https://github.com/facebook/react/blob/204a551eae466ab74ba23870f61dc2b5c71d5ab2/packages/react-devtools-shared/src/backend/agent.js#L219

Reply:
Talked irl. What I was asking was

1. ProfilerStore subscribes to bridge
2. Agent subscribes to bridge
3. Bridge emits shutdown event
4. Agent remove all listeners here
5. Would ProfileStore get the shutdown event?

We found that the current impl does account for this (removing all listeners won't prevent existing listeners from getting the currently-processed event):

https://github.com/facebook/react/blob/d5bba18b5d81f234657586865248c5b6849599cd/packages/react-devtools-shared/src/events.js#L42

We should prob still be a good neighbour and only unsubscribe our own listeners tho